src/actions/TemperatureSyncActions.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import { generateUUID } from 'react-native-database';
+
+import { UIDatabase } from '../database';
+import { Sensor } from '../database/DataTypes';
+
+export const TEMPERATURE_SYNC_ACTIONS = {
+  SCAN_START: 'TemperatureSync/scanStart',
+  SCAN_COMPLETE: 'TemperatureSync/scanComplete',
+  SCAN_ERROR: 'TemperatureSync/scanError',
+};
+
+const scanStart = () => ({ type: TEMPERATURE_SYNC_ACTIONS.SCAN_START });
+const scanComplete = () => ({ type: TEMPERATURE_SYNC_ACTIONS.SCAN_COMPLETE });
+const scanError = () => ({ type: TEMPERATURE_SYNC_ACTIONS.SCAN_ERROR });
+
+const updateSensors = sensorAdvertisements => dispatch => {
+  const isArray = Array.isArray(sensorAdvertisements);
+
+  if (!isArray) dispatch(scanError());
```

Comment:
I like early returns. I would do this:

```suggestion
  if (!isArray) {
    dispatch(scanError());
    return;
 }
```